jrugged-core/src/test/java/org/fishwife/jrugged/TestSkepticBreaker.java

```diff
@@ -0,0 +1,543 @@
+
+        Object theReturned = impl.invoke(mockRunnable, result);
+
+        verify(mockRunnable);
+        assertSame(result, theReturned);
+    }
+
+    @Test
+    public void testInvokeWithRunnableResultAndByPassReturnsResult() throws Exception {
+        final Object result = new Object();
+        impl.setByPassState(true);
+
+        mockRunnable.run();
+        replay(mockRunnable);
+
+        Object theReturned = impl.invoke(mockRunnable, result);
+
+        verify(mockRunnable);
+        assertSame(result, theReturned);
+    }
+
+    @Test(expected = BreakerException.class)
+    public void testInvokeWithRunnableResultAndTripHardReturnsException() throws Exception {
+        final Object result = new Object();
+        impl.tripHard();
+
+        mockRunnable.run();
+        replay(mockRunnable);
+
+        impl.invoke(mockRunnable, result);
+
+        verify(mockRunnable);
+    }
+
+    @Test
+    public void testInvokeWithRunnableDoesNotError() throws Exception {
+        mockRunnable.run();
+        replay(mockRunnable);
+
+        impl.invoke(mockRunnable);
+
+        verify(mockRunnable);
+    }
+
+    @Test
+    public void testInvokeWithRunnableAndByPassDoesNotError() throws Exception {
+        impl.setByPassState(true);
+
+        mockRunnable.run();
+        replay(mockRunnable);
+
+        impl.invoke(mockRunnable);
+
+        verify(mockRunnable);
+    }
+
+    @Test(expected = BreakerException.class)
+    public void testInvokeWithRunnableAndTripHardReturnsException() throws Exception {
+        impl.tripHard();
+
+        mockRunnable.run();
+        replay(mockRunnable);
+
+        impl.invoke(mockRunnable);
+
+        verify(mockRunnable);
+    }
+
+    @Test
+    public void testStaysClosedOnSuccess() throws Exception {
+        impl.state = SkepticBreaker.BreakerState.CLOSED;
+        final Object obj = new Object();
+        expect(mockCallable.call()).andReturn(obj);
+        replay(mockCallable);
+
+        Object result = impl.invoke(mockCallable);
+
+        verify(mockCallable);
+        assertSame(obj, result);
+        assertEquals(SkepticBreaker.BreakerState.CLOSED, impl.state);
+    }
+
+    @Test
+    //Add check on timers here as well
+    public void testOpensOnFailure() throws Exception {
+        long start = System.currentTimeMillis();
+        impl.state = SkepticBreaker.BreakerState.OPEN;
+        expect(mockCallable.call()).andThrow(new RuntimeException());
+        replay(mockCallable);
+
+        try {
+            impl.invoke(mockCallable);
+            fail("should have thrown an exception");
+        } catch (RuntimeException expected) {
+        }
+
+        long end = System.currentTimeMillis();
+
+        verify(mockCallable);
+        assertEquals(SkepticBreaker.BreakerState.OPEN, impl.state);
+        assertTrue(impl.lastFailure.get() >= start);
+        assertTrue(impl.lastFailure.get() <= end);
+    }
+
+    @Test
+    public void testOpenDuringCooldownThrowsCBException()
```

Comment:
The name of this test is wrong now, it isn't a CBException it is just a BreakerException.